src/contracts/strategies/StrategyFactory.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "./StrategyFactoryStorage.sol";
+import "./StrategyBase.sol";
+import "../permissions/Pausable.sol";
+
+/**
+ * @title Factory contract for deploying BeaconProxies of a Strategy contract implementation for arbitrary ERC20 tokens
+ *        and automatically adding them to the StrategyWhitelist in EigenLayer.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @dev This may not be compatible with non-standard ERC20 tokens. Caution is warranted.
+ */
+contract StrategyFactory is StrategyFactoryStorage, OwnableUpgradeable, Pausable {
+    uint8 internal constant PAUSED_NEW_STRATEGIES = 0;
+
+    /// @notice EigenLayer's StrategyManager contract
+    IStrategyManager public immutable strategyManager;
+
+    /// @notice Since this contract is designed to be initializable, the constructor simply sets the immutable variables.
+    constructor(IStrategyManager _strategyManager) {
+        strategyManager = _strategyManager;
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _initialOwner,
+        IPauserRegistry _pauserRegistry,
+        uint256 _initialPausedStatus,
+        IBeacon _strategyBeacon
+    ) public virtual initializer {
+        _transferOwnership(_initialOwner);
+        _initializePauser(_pauserRegistry, _initialPausedStatus);
+        _setStrategyBeacon(_strategyBeacon);
+    }
+
+    /**
+     * @notice Deploy a new strategyBeacon contract for the ERC20 token.
+     * @dev A strategy contract must not yet exist for the token.
+     * $dev Immense caution is warranted for non-standard ERC20 tokens, particularly "reentrant" tokens
+     * like those that conform to ERC777.
+     */
+    function deployNewStrategy(IERC20 token)
+        external
+        onlyWhenNotPaused(PAUSED_NEW_STRATEGIES)
+        returns (IStrategy newStrategy)
+    {
+        require(!isBlacklisted[token], "StrategyFactory.deployNewStrategy: Token is blacklisted");
+        require(
+            tokenStrategy[token] == IStrategy(address(0)),
+            "StrategyFactory.deployNewStrategy: Strategy already exists for token"
+        );
+        IStrategy strategy = IStrategy(
+            address(
+                new BeaconProxy(
+                    address(strategyBeacon),
+                    abi.encodeWithSelector(StrategyBase.initialize.selector, token, pauserRegistry)
+                )
+            )
+        );
```

Comment:
Should we be using Create2 like we do in the EigenPodManager?

Reply:
As an outsider, I definitely +1 this